src/tests/ftest/util/dmg_utils.py

```diff
@@ -23,44 +23,190 @@
 """
 from __future__ import print_function
 
-from command_utils import ExecutableCommand
+import os
+import getpass
+
+from command_utils import DaosCommand, CommandWithParameters
 from command_utils import BasicParameter, FormattedParameter
+from general_utils import get_file_path
+from avocado.utils import process
 
 
-class DmgCommand(ExecutableCommand):
+class DmgCommand(DaosCommand):
     """Defines a object representing a dmg (or daos_shell) command."""
 
     def __init__(self, path):
         """Create a dmg Command object."""
         super(DmgCommand, self).__init__("/run/dmg/*", "daos_shell", path)
+        self.format = self.DmgFormatSubCommand()
+        self.prepare = self.DmgPrepareSubCommand()
 
-        self.request = BasicParameter("{}")
-        self.action = BasicParameter("{}")
-
-        # daos_shell options
         self.hostlist = FormattedParameter("-l {}")
         self.hostfile = FormattedParameter("-f {}")
         self.configpath = FormattedParameter("-o {}")
         self.insecure = FormattedParameter("-i", None)
+        self.debug = FormattedParameter("-d", None)
+        self.json = FormattedParameter("-j", None)
+        self.subcommand_list = {
+            "format": self.format,
+            "prepare": self.prepare,
+        }
+
+    def __str__(self):
+        """Return the command with all of its defined parameters as a string.
+
+        Returns:
+            str: the command with all the defined parameters
+
+        """
+        # Join all the parameters that have been assigned a value with the
+        # command to create the command string
+        params = []
+        for name in self.get_param_names():
+            value = str(getattr(self, name))
+            if name == "action" and value in self.subcommand_list:
+                params.append(str(self.subcommand_list[value]))
+            elif value != "":
+                params.append(value)
+        return " ".join([os.path.join(self._path, self._command)] + params)
+
+    class DmgFormatSubCommand(CommandWithParameters):
+        """Defines a object representing a sub dmg (or daos_shell) command."""
+
+        def __init__(self):
+            """Create a dmg Command object."""
+            super(DmgCommand.DmgFormatSubCommand, self).__init__("format")
+            self.force = FormattedParameter("-f", None)
+
+    class DmgPrepareSubCommand(CommandWithParameters):
+        """Defines a object representing a sub dmg (or daos_shell) command."""
```

Comment:
Specifically isn't this the `prepare` dmg sub command?

Reply:
Yes, I'll specify.